client/components/ManageProducts/EditProduct.js

```diff
@@ -0,0 +1,144 @@
+    this.onSave = this.onSave.bind(this);
+    this.getCurrProduct = this.getCurrProduct.bind(this);
+  }
+
+  componentDidMount() {
+    this.getCurrProduct();
+  }
+
+  onChange(ev) {
+    const change = {};
+    change[ev.target.name] = ev.target.value;
+    this.setState(change);
+  }
+
+  async onSave(ev) {
+    const { history, update } = this.props;
+    ev.preventDefault();
+    const { id } = this.props.match.params;
+    try {
+      const {
+        title,
+        brand,
+        description,
+        price,
+        inventory,
+        imageUrl,
+        reqCountry,
+      } = this.state;
+      update({
+        id,
+        title,
+        brand,
+        description,
+        price,
+        inventory,
+        imageUrl,
+        reqCountry,
+      });
+
+      history.push('/manage-products');
+    } catch (error) {
+      this.setState({ error: error.response.data.error });
+    }
+  }
+
+  async getCurrProduct() {
+    const { id } = this.props.match.params;
+    const currProduct = (await axios.get(`/api/products/${id}`)).data;
+    const {
+      title,
+      brand,
+      description,
+      price,
+      inventory,
+      imageUrl,
+      reqCountry,
+    } = currProduct;
+    this.setState({
+      title,
+      brand,
+      description,
+      price,
+      inventory,
+      imageUrl,
+      reqCountry,
+    });
+  }
+
+  render() {
+    const {
+      title,
+      brand,
+      description,
+      price,
+      inventory,
+      imageUrl,
+      reqCountry,
+    } = this.state;
+    const { onChange, onSave } = this;
+    return (
+      <div id="edit-product">
+        <h3>Edit Product:</h3>
+        <img src={imageUrl} alt="snack" width="100" />
+        <form onSubmit={onSave}>
+          <label>Title*:</label>
+          <input name="title" value={title} onChange={onChange} />
+          <br />
+          <label>Brand*:</label>
+          <input name="brand" value={brand} onChange={onChange} />
+          <br />
+          <label>Description*:</label>
+          <input name="description" value={description} onChange={onChange} />
+          <br />
+          <label>Price:*</label>
+          <input name="price" value={price} onChange={onChange} />
+          <br />
+          <label>Inventory*:</label>
+          <input
+            name="inventory"
+            value={inventory}
+            type="number"
+            onChange={onChange}
+          />
+          <br />
+          <label>Image Url*:</label>
+          <input name="imageUrl" value={imageUrl} onChange={onChange} />
+          <br />
+          <label>Country*:</label>
+          <input name="reqCountry" value={reqCountry} onChange={onChange} />
+          <br />
+          <button>Submit Changes</button>
+        </form>
+      </div>
+    );
+  }
+}
+
+const mapDispatchToProps = (dispatch, { history }) => {
+  return {
+    update: (updatedProduct) =>
+      dispatch(updateProduct(updatedProduct, history)),
+  };
+};
+
```

Comment:
Just a general note, again something I can go in and edit when I add to this but reqCountry wouldn't work here because its an included model. You would need to do something like product.country.

Reply:
To edit country, it might be easier to do a drop down menu so we have all the countryIds since that's was is on the Product model.

Reply:
Instead of passing reqCountry, it should be countryId. Again, I can change al this when I add all the Category stuff to it.

Reply:
Thanks, Alejandra. I'll make a note to add the dropdown in. And I should have left more of an explanation on why I used 'reqCountry' instead of 'country'. When I used 'country' in the CreateProduct component's state, that didn't work because the post route for countries was using 'location '. Then I tried to use location, but that didn't work because this.state.location was providing the current url instead of the country's name. I hesitated to change anything up inside the route, but in the end it seemed like just changing the naming of the variable inside the products post route from 'location' to 'reqCountry' was the lowest-impact way to get the post request to go through and get a new product created, rather than messing with anything else within the post route. The product creation was working and all the tests were passing so I figured I didn't throw anything off by changing the naming a bit. But definitely let me know if you still think this will create issues.

Reply:
And I'll wait until your PR is merged before I merge this

Reply:
So this will work for POST creations because of the way we have the route set up but  it wouldn't work for the Put request since the put route isnt set up that way. If you have a drowndown menu with all the countries already there, instead of reqCountry, we can send in the countryId and it would work for both (you'd just need to change the post route a tiny bit) or you can change the put route to work the way the post route works. This will definetely break some test specs though;